internal/database/permission_sync_jobs.go

```diff
@@ -131,11 +145,117 @@ func (s *permissionSyncJobStore) createSyncJob(ctx context.Context, job *Permiss
 	return scanPermissionSyncJob(job, s.QueryRow(ctx, q))
 }
 
+// checkDuplicateAndCreateSyncJob adds a new perms sync job if there is no
+// present duplicate of it.
+//
+// Duplicates are handled in this way:
+//
+// 1) If there is no existing job for given user/repo ID in a queued state, we
+// insert right away.
+//
+// 2) If there is an existing job with lower priority, we cancel it and insert a
+// new one with higher priority.
+//
+// 3) If there is an existing job with higher priority, we don't insert new job.
+func (s *permissionSyncJobStore) checkDuplicateAndCreateSyncJob(ctx context.Context, job *PermissionSyncJob) (err error) {
+	tx, err := s.Transact(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = tx.Done(err)
+	}()
+	opts := ListPermissionSyncJobOpts{UserID: job.UserID, RepoID: job.RepositoryID, State: "queued", NotCanceled: true, NullProcessAfter: true}
+	syncJobs, err := tx.List(ctx, opts)
+	if err != nil {
+		return err
+	}
+	// Job doesn't exist -- create it
+	if len(syncJobs) == 0 {
+		return s.create(ctx, job)
```

Comment:
This is a bug because you're doing the `s.create` outside the transaction. What you need is to pass `tx` to `s.create` so it's created inside the transaction and then committed when the outer function, `checkDuplicateAndCreateSyncJob`, returns.

```suggestion
		return tx.create(ctx, job)
```

this should do it, I think